spec/services/charges/charge_models/volume_service_spec.rb

```diff
@@ -33,48 +33,96 @@
+        {
+          flat_amount: 0,
+          per_unit_amount: 0,
+          per_unit_total_amount: 0,
+        }
+      )
     end
   end
 
   context 'when aggregation is 1' do
     let(:aggregation) { 1 }
 
-    it 'applies a unit amount for 1 and the flat amount' do
+    it 'applies a unit amount for 1 and the flat amount', :aggregate_failures do
       expect(apply_volume_service.amount).to eq(12)
+      expect(apply_volume_service.unit_amount).to eq(12)
+      expect(apply_volume_service.amount_details).to eq(
+        {
+          flat_amount: 10,
+          per_unit_amount: 2,
+          per_unit_total_amount: 2,
+        }
+      )
     end
   end
 
   context 'when aggregation is the limit of the first range' do
     let(:aggregation) { 100 }
 
-    it 'applies unit amount for the first range and the flat amount' do
+    it 'applies unit amount for the first range and the flat amount', :aggregate_failures do
       expect(apply_volume_service.amount).to eq(210)
+      expect(apply_volume_service.unit_amount).to eq(2.1)
+      expect(apply_volume_service.amount_details).to eq(
+        {
+          flat_amount: 10,
+          per_unit_amount: 2,
+          per_unit_total_amount: 200,
+        }
+      )
     end
   end
 
   context 'when aggregation is the lower limit of the second range' do
     let(:aggregation) { 101 }
 
-    it 'applies unit amount the second range and no flat amount' do
+    it 'applies unit amount the second range and no flat amount', :aggregate_failures do
       expect(apply_volume_service.amount).to eq(101)
+      expect(apply_volume_service.unit_amount).to eq(1)
+      expect(apply_volume_service.amount_details).to eq(
+        {
+          flat_amount: 0,
+          per_unit_amount: 1,
+          per_unit_total_amount: 101,
+        }
+      )
     end
   end
 
   context 'when aggregation is the uper limit of the second range' do
     let(:aggregation) { 200 }
 
-    it 'applies unit amount the second range and no flat amount' do
+    it 'applies unit amount the second range and no flat amount', :aggregate_failures do
       expect(apply_volume_service.amount).to eq(200)
+      expect(apply_volume_service.unit_amount).to eq(1)
+      expect(apply_volume_service.amount_details).to eq(
+        {
+          flat_amount: 0,
+          per_unit_amount: 1,
+          per_unit_total_amount: 200,
+        }
+      )
     end
   end
 
   context 'when aggregation is the above the lower limit of the last range' do
     let(:aggregation) { 300 }
 
-    it 'applies unit amount the second range and no flat amount' do
+    it 'applies unit amount the second range and no flat amount', :aggregate_failures do
       expect(apply_volume_service.amount).to eq(200)
+      expect(apply_volume_service.unit_amount.round(2)).to eq(0.67)
+      expect(apply_volume_service.amount_details).to eq(
+        {
+          flat_amount: 50,
+          per_unit_amount: 0.5,
+          per_unit_total_amount: 150,
+        }
```

Comment:
Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)